ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/accordion/v1/accordion/clientlibs/site/js/accordionview.js

```diff
@@ -63,6 +138,531 @@
+         *
+         * @private
+         * @param {HTMLElement} wrapper The Accordion wrapper element
+         */
+        #cacheElements(wrapper) {
+            this._elements = {};
+            this._elements.self = wrapper;
+            var hooks = this._elements.self.querySelectorAll("[data-" + Accordion.NS + "-hook-" + Accordion.IS + "]");
+
+            for (var i = 0; i < hooks.length; i++) {
+                var hook = hooks[i];
+                if (hook.closest("[data-cmp-is=" + Accordion.IS + "]") === this._elements.self) { // only process own accordion elements
+                    var lowerCased = Accordion.IS.toLowerCase();
+                    var capitalized = lowerCased.charAt(0).toUpperCase() + lowerCased.slice(1);
+                    var key = hook.dataset[Accordion.NS + "Hook" + capitalized];
+                    if (this._elements[key]) {
+                        if (!Array.isArray(this._elements[key])) {
+                            var tmp = this._elements[key];
+                            this._elements[key] = [tmp];
+                        }
+                        this._elements[key].push(hook);
+                    } else {
+                        this._elements[key] = [hook];
+                    }
+                }
+            }
+        }
+
+        /**
+         * Returns all expanded items.
+         *
+         * @private
+         * @returns {HTMLElement[]} The expanded items
+         */
+        #getExpandedItems() {
+            var expandedItems = [];
+
+            for (var i = 0; i < this.#getCachedItems().length; i++) {
+                var item = this.#getCachedItems()[i];
+                var expanded = this.#isItemExpanded(item);
+                if (expanded) {
+                    expandedItems.push(item);
+                }
+            }
+
+            return expandedItems;
+        }
+
+        /**
+         * Gets an item's expanded state.
+         *
+         * @private
+         * @param {HTMLElement} item The item for checking its expanded state
+         * @returns {Boolean} true if the item is expanded, false otherwise
+         */
+        #isItemExpanded(item) {
+            return item && item.dataset && item.dataset["cmpExpanded"] !== undefined;
+        }
+
+        /**
+         * Binds Accordion event handling.
+         *
+         * @private
+         */
+        #bindEvents() {
+            var buttons = this.#getCachedButtons();
+            if (buttons) {
+                var _self = this;
+                for (var i = 0; i < buttons.length; i++) {
+                    (function (index) {
+                        buttons[index].addEventListener("click", function (event) {
+                            var itemDivId = _self.#convertToItemDivId(buttons[index].id);
+                            _self.#toggle(itemDivId);
+                            _self.#collapseAllOtherItems(itemDivId);
```

Comment:
how is collapseAllOtherItems related to repeatability support? are we diverging from sites accordion here?

Reply:
Hi Nitin,
We are not diverging in any way, the functionality of the forms accordion remains the same.

In sites accordion they have kept a variable in code named properties which has a key isSingleExpansion, which is by default set as false.

In forms accordion html we have hardcoded that as true because even AF1 forms accordion behaved like only one accordion tab expanded at a time

https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/accordion/v1/accordion/accordion.html#L29

Using that variable they had implemented the same logic of collapsingAllOtherItems inside toggle method which isn't correct as it violates the single responsibility principle and makes the code less readable
I have just extracted out that logic and have added it in a prvt method

reference of the site code
https://github.com/adobe/aem-core-wcm-components/blob/main/content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js#L330